src/TheGuardianIn2020Banner/index.tsx

```diff
@@ -27,8 +27,25 @@ const catchAndLogErrors = (description: string, fn: () => void): void => {
 };
 
 export const COMPONENT_NAME = 'TheGuardianIn2020Banner';
-const THE_GU_IN_2020_URL =
-    'https://www.theguardian.com/info/ng-interactive/2020/dec/21/the-guardian-in-2020?utm_source=eml&utm_medium=embau&utm_campaign=MK_SU_WorkingReport2020Canvas&CMP_TU=mrtn&CMP_BUNIT=supporters&INTCMP=Working_Report_2020_Retention_Banner';
+
+const urlObject = new URL(
+    'https://www.theguardian.com/info/ng-interactive/2020/dec/21/the-guardian-in-2020',
+);
+
+const urlParams = {
+    utm_source: 'eml',
+    utm_medium: 'embau',
+    utm_campaign: 'MK_SU_WorkingReport2020Canvas',
+    CMP_TU: 'mrtn',
+    CMP_BUNIT: 'supporters',
+    INTCMP: 'Working_Report_2020_Retention_Banner',
+};
+
+for (const [key, value] of Object.entries(urlParams)) {
+    urlObject.searchParams.append(key, value);
```

Comment:
I feel `.set` would be more suitable than `.append` as it would ensure there are no repeat parameters.

See: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/set

Reply:
Agree with this comment about using `set` :+1:

Purely personal preference and totally something to take or leave, but whenever I see a list of things being condensed down into a single thing like this I think [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce).

Reply:
Thanks both, have updated to use `.set()`.

I had a go using reduce, and I think the solution here would be - 
```
Object.entries(urlParams).reduce((acc, cur) => {
    const [key, value] = cur;
    acc.searchParams.set(key, value);
    return acc;
}, urlObject);
```
- which is a bit more verbose than the `for ... of` loop and I'm not sure it's more readable in this case (though perhaps my implementation is a bit clunky). Also worth noting that we're mutating the same URL object repeatedly (via `.set`) rather than actually passing forward a 'new' accumulator value each time which feels a bit misleading.

Reply:
Agreed, mutating in a reduce is a bit misleading. We'd probably want to refactor that to be a more conventional accumulator if we were going to use reduce. OK as-is though.

Reply:
Hey sorry I missed your update Rhys. Ah yep now you've written it out I agree with Adam, the combination of mutating and reduce isn't great. Sorry for the derailment!